### DISCUSSION TITLE
Is it possible to set the "name" and "category" of synteny tracks according to the current linear genome view?

### DISCUSSION THREAD

**Discussion by carolyncaron** [GMOD/jbrowse-components#4324] [Q&A]
This is an example of the current synteny track listing for one of our genomes (order is reference genome vs. query genome): 

<img width="202" alt="Screen Shot 2024-03-27 at 11 51 44 AM" src="https://github.com/GMOD/jbrowse-components/assets/7927760/bad7ac8a-b119-42e7-adb0-e679804ab4cb">

I want to sub-categorize the tracks to make it easier to find the right synteny track, and distinguish the difference between reference and query. We are anticipating this list to get a bit longer as well. I was hoping to split the list into 2 sub-categories: 
1) The current genome assembly as reference
2) The current genome assembly as the query

Then the track names could be set to simply the name of the other assembly. I altered my config.json to accomplish this for one of our genomes to demonstrate this:
<img width="356" alt="Screen Shot 2024-03-27 at 12 31 24 PM" src="https://github.com/GMOD/jbrowse-components/assets/7927760/6babf8a7-7b9c-4d75-a70f-0742654ebad7">

But sadly, since synteny tracks in the config get added to both linear genome views, any synteny tracks involving Lcu.1GRN have already been renamed and categorized. Thus, I'd be adding additional categories for the other genomes and the simplicity gets lost. I hope this makes sense, let me know if you need more detail!

I read through the docs, thinking this could maybe be accomplished using a jexl callback, but all the examples I could find involve features, so I don't know if it is appropriate here. Do I instead need a no-build plugin? Am I right to assume that the extension point Core-preProcessTrackConfig would apply here? Any advice is appreciated!

**Comment by cmdcolin** [comment]
I agree that it would be good to have something to aid this situation. I'm working on creating a large demo instance as well and saw a similar thing occuring. unfortunately the track selector doesnt have any jexl callback for displaying the track label or categories. I will ponder this a bit...the best scenario would be for this to happen automatically so people don't have to have a custom advanced config to solve it properly, so i'm wondering if some auto-categorization could be added for tracks of type SyntenyTrack perhaps...

**Reply by carolyncaron** [reply]
Thanks for your reply! Yes, some auto categorization could be great. I will say that the current functionality is fine though if categorization is on something other than by genome assembly. For example, to distinguish between software used. We did this for one of our synteny tracks and it was nice to see the category was added to the track listings for both genomes 😅
